astropy/table/pprint.py

```diff
@@ -482,32 +459,32 @@ def _pformat_table(self, table, max_lines=None, max_width=None,
         if show_unit is None:
             show_unit = any([col.info.unit for col in six.itervalues(table.columns)])
 
-        # Figure out align
-        if isinstance(align, str):
-            align = align
-        elif isinstance(align, list) and len(align) == 1:
-            align = align[0]
-        elif isinstance(align, list) and len(align) == len(table.columns.values()):
-            align = align
-        else:
-            align = 'right'
-
-        # If align remains a list, need to loop over values
-        if type(align) == list:
-            for i, col in enumerate(table.columns.values()):
-                lines, outs = self._pformat_col(col, max_lines, show_name=show_name,
-                                                show_unit=show_unit, show_dtype=show_dtype,
-                                                align=align[i])
-                if outs['show_length']:
-                    lines = lines[:-1]
-                cols.append(lines)
+        # Coerce align into a correctly-sized list of alignments (if possible)
+        n_cols = len(table.columns)
+        if align is None:
+            align = [None] * n_cols
+
+        if isinstance(align, six.string_types):
```

Comment:
For this case where one gives an alignment that holds for all columns, I feel it would be best if it did *not* override any alignment given in the column format. But probably that is a bit too tricky to implement...